woocommerce/wc-admin
# Update UI for Payment Task Screen

Fixes #6683 

Implementing new UI for payment task. 

**Outstanding questions**
- I'm a little uncertain of how to handle the image assets, namely the WCPay logo & credit card images. After asking around I was pointed towards [these assets already in the woocommerce repo](https://github.com/woocommerce/woocommerce/tree/23710744c01ded649d6a94a4eaea8745e543159f/assets/images/icons/credit-cards), but they're not all there, those that are have a different appearance, and I'm not sure of the pattern for referencing assets in another plugin.
 Right now I have them all in their own SVG files, although that of course drives up the plugin size a bit. I can also convert the CC icons to PNG files if desired, since those would be significantly lighter in that case (if we don't intent to use these in larger sizes elsewhere).

**WCPay not enabled**
![image](https://user-images.githubusercontent.com/444632/114212199-66ebed00-9916-11eb-8d12-45702da6da2f.png)

**WCPay enabled but not configured**
![image](https://user-images.githubusercontent.com/444632/114212353-96025e80-9916-11eb-956c-f9491fdbe418.png)

**WCPay enabled & configured**
![image](https://user-images.githubusercontent.com/444632/114212439-b2060000-9916-11eb-9baf-0d1eb0be5e7d.png)

**Mobile view**
![image](https://user-images.githubusercontent.com/444632/114212646-ec6f9d00-9916-11eb-9216-c2b4d84b963c.png)


**Acceptance criteria:** 

- [x] Implement the new layout + copy
- [x] "Learn more" should link to: https://woocommerce.com/payments/
- [x] Clicking "Get started" will redirect users to the WCPayments KYC flow (as we have right now)
- [x] All other payment methods are located in a separate card titled "Additional payment methods".
- [x] Update the WCPayments card to the layout displayed above when the user completes the setup
- [x] Update the WCPayments card with the following if the user visits the payments task after installing the extension but without completing the setup:
Switch the "recommended" pill to "Setup required"
Update the CTA copy to "Finish setup"
([mockup](https://github.com/woocommerce/woocommerce-admin/issues/6683#issuecomment-811173292) in this comment)
- [x] Redirect to the WCPayments dashboard when the user clicks "Manage"

### Accessibility




### Screenshots

### Detailed test instructions:

-  Create fresh site and install plugin(s).
- Navigate to Homescreen & click "Choose payment methods" task.
- You should see the UI from the initial screenshot, with the WCPay Card including the "recommended" pill and "Get started" button.
- Ensure "Learn more" link and "Get started" buttons work as per the acceptance criteria. 
- Confirm `wcadmin_tasklist_payment_learn_more` tracks event is fired while clicking "Learn more."
- Check mobile view (last screenshot).
- Enable WCPay, but don't configure (I do this by overriding boolean values in `methods.js`
- View screen again, which should now appear like the second screenshot, including "Setup required" icon & "Finish setup" button.
- Configure WCPay (again, I do this through overriding boolean in `methods.js`).
- Now you should see WCPay appear in "Enabled payment methods" card, as per the third screenshot.
- Test "Manage" button for WCPay, and ensure `wcadmin_tasklist_payment_manage` tracks event is fired.
- Check mobile view again.
- Quick smoke test that includes the rest of the functionality on this page, including "Set up" buttons for other payment methods.
